src/mapblock.cpp

```diff
@@ -102,11 +102,9 @@ static const char *modified_reason_strings[] = {
 MapBlock::MapBlock(v3s16 pos, IGameDef *gamedef):
 		m_pos(pos),
 		m_pos_relative(pos * MAP_BLOCKSIZE),
-		data(new MapNode[nodecount]),
 		m_gamedef(gamedef)
 {
-	reallocate();
-	assert(m_modified > MOD_STATE_CLEAN);
+	reallocate(nodecount, MapNode(CONTENT_IGNORE));
 }
 
 MapBlock::~MapBlock()
```

Comment:
the destructor contains code that needs to be adjusted

Reply:
Oh, the size tracking, yes.

Reply:
I would simply put the `TrackFreedMemory` call behind `if (!m_is_monoblock)`